server/app/services/mongodb/migrator.rb

```diff
@@ -68,13 +69,16 @@ def migrations
     def migrate
       ensure_indexes
       release_stale_lock
-      with_dlock(LOCK_NAME, LOCK_TIMEOUT) do
-        migrate_without_lock
-      end
+      lock_id = wait_until!("lock #{name} is available", timeout: LOCK_TIMEOUT, interval: 0.5) { 
+        DistributedLock.obtain_lock(LOCK_NAME)
+      }
+      migrate_without_lock
+    ensure
+      DistributedLock.release_lock(LOCK_NAME, lock_id)
```

Comment:
Needs an `if lock_id` in case the wait times out, and there is no `lock_id` set?